src/main/java/com/staxrt/tutorial/services/CarService.java

```diff
@@ -1,32 +1,45 @@
 package com.staxrt.tutorial.services;
 
 import com.staxrt.tutorial.model.Car;
+import com.staxrt.tutorial.repository.CarRepository;
 import lombok.RequiredArgsConstructor;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.jdbc.core.BeanPropertyRowMapper;
-import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
+
 import java.util.List;
+import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class CarService {
-    @Autowired
-    JdbcTemplate jdbcTemplate;
+
+    private final CarRepository carRepository;
+
     public List<Car> getAllCars() {
-        String sql = "SELECT * FROM cars";
-        return (List<Car>) jdbcTemplate.query(sql,new BeanPropertyRowMapper(Car.class));
+        return carRepository.findAll();
     }
-    public void createCar(Car car){
-        String sql = "INSERT INTO cars (car_brand,engine_capacity,vin) VALUES (?,?,?)";
-        jdbcTemplate.update(sql,car.getCarBrand(),car.getEngineCapacity(),car.getVin());
+
+    public void createCar(Car car) {
+        carRepository.save(car);
     }
-    public void deleteCar(Long id){
-        String sql = "DELETE FROM cars WHERE id="+id;
-        jdbcTemplate.execute(sql);
+
+    public ResponseEntity deleteCar(Long id) {
+        Optional<Car> car = carRepository.findById(id);
+        if (car.isPresent()) {
+            carRepository.delete(car.get());
+            return ResponseEntity.ok().build();
+        } else {
+            return ResponseEntity.notFound().build();
+        }
     }
-    public Car findCarById(Long id){
-        String sql = "SELECT * FROM cars WHERE id="+id;
-        return (Car) jdbcTemplate.queryForObject(sql, new Object[]{}, new BeanPropertyRowMapper(Car.class));
+
+    public ResponseEntity findCarById(Long id) {
+        Optional<Car> car = carRepository.findById(id);
+        if (car.isPresent()) {
+            return ResponseEntity.ok().body(car.get());
```

Comment:
ResponseEntity should be only used in Controller layer. Here return Car object or data trasnfer objects in the future.